src/Faker/Provider/ms_MY/Address.php

```diff
@@ -0,0 +1,1340 @@
+<?php
+
+namespace Faker\Provider\ms_MY;
+
+use \Faker\Generator;
+
+class Address extends \Faker\Provider\Address
+{
+    /**
+     * @link https://en.wikipedia.org/wiki/Addresses_in_Malaysia
+     */
+    protected static $addressFormats = array(
+        '{{streetAddress}}, {{township}}, {{townState}}',
+    );
+
+    protected static $streetAddressFormats = array(
+        '{{buildingPrefix}}{{buildingNumber}}, {{streetName}}'
+    );
+
+    /**
+     * Most of the time 'No.' is not needed, and 'Lot' is less used.
+     */
+    protected static $buildingPrefix = array(
+        '','','','','','',
+        'No. ','No. ','No. ',
+        'Lot ',
+    );
+
+    protected static $buildingNumber = array(
+        '%','%','%',
+        '%#','%#','%#','%#',
+        '%##',
+        '%-%',
+        '?-##-##',
+        '%?-##'
+    );
+
+    protected static $streetNameFormats = array(
+        '{{streetPrefix}} %',
+        '{{streetPrefix}} %/%',
+        '{{streetPrefix}} %/%#',
+        '{{streetPrefix}} %/%?',
+        '{{streetPrefix}} %/%#?',
+        '{{streetPrefix}} %?',
+        '{{streetPrefix}} %#?',
+        '{{streetPrefix}} {{streetSuffix}}',
+        '{{streetPrefix}} {{streetSuffix}} %',
+        '{{streetPrefix}} {{streetSuffix}} %/%',
+        '{{streetPrefix}} {{streetSuffix}} %/%#',
+        '{{streetPrefix}} {{streetSuffix}} %/%?',
+        '{{streetPrefix}} {{streetSuffix}} %/%#?',
+        '{{streetPrefix}} {{streetSuffix}} %?',
+        '{{streetPrefix}} {{streetSuffix}} %#?',
+    );
+
+    protected static $townshipFormats = array(
```

Comment:
there is way too much data on this provideR. Faker doesn't look to be exhaustive, or to copy the geographic census of every country. Please provide at most 300 township names, city names, and street names.

Reply:
township names, city names, and street names data greatly reduced in latest commit.

Reply:
Still too much granularity. I suggest the class only exposes one `city` formatter, no need to expose `townJohor` etc as public methods. And that means you can use an associative array to define the data for cities and townships and postcodes in a more concise way.

Reply:
Suggestion taken. The data are now in associative arrays with single public method for each.